### PR TITLE
feat: Implements `mongodbatlas_encryption_at_rest_private_endpoint` singular data source

### PR DESCRIPTION
**Pull request by AgustinBettati**
## Description

Link to any related issue(s): CLOUDP-267660

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [x] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals I have added appropriate changelog entries.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

**Comment by AgustinBettati** [comment]
With the approach of having the schemas/conversion methods defined in a separate PR its nice to see how small these followup PRs are getting @maastha

File: internal/testutil/acc/attribute_checks.go [hunk 1]
```diff
@@ -65,6 +65,24 @@ func JSONEquals(expected string) resource.CheckResourceAttrWithFunc {
 	}
 }
 
+// CheckRSAndDS returns a check function that asserts a set of attributes (presence and values) in resource and data sources.
+// If a plural data source name is provided, it will apply checks over first result
+func CheckRSAndDS(resourceName string, dataSourceName, pluralDataSourceName *string, attrsSet []string, attrsMap map[string]string, extra ...resource.TestCheckFunc) resource.TestCheckFunc {
```
Comment (EspenAlbert) [inline internal/testutil/acc/attribute_checks.go:70]: Nice 🤩

**Review by EspenAlbert — APPROVED** [review]
LGTM!

**Review by maastha — APPROVED** [review]
>With the approach of having the schemas/conversion methods defined in a separate PR its nice to see how small these followup PRs are getting @maastha

LGTM! and totally agree!

**Review by lmkerbey-mdb — COMMENTED** [review]
LGTM.

**Pull request merged**